build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+
+                        echo "Cleaning test result files 'TestResult-*.xml'"
+                        cleanFiles("TestResult-*.xml")
+
+                        // Avoid: EXEC : error : jnimarshalmethod-gen: Unable to create Java VM
+                        def XAInstallPrefix = "bin/${env.BuildFlavor}/lib/xamarin.android"
+
+                        def javaInteropDllConfigFile = "${XAInstallPrefix}/xbuild/Xamarin/Android/Java.Interop.dll.config"
+                        if (fileExists(javaInteropDllConfigFile)) {
+                            echo "Cleansing file '${javaInteropDllConfigFile}'"
+                            sh "rm -f ${javaInteropDllConfigFile}"
+                        }
+
+                        def javaEnvironmentConfigFile = "${XAInstallPrefix}/xbuild/Xamarin/Android/Java.Runtime.Environment.dll.config"
+                        if (fileExists(javaEnvironmentConfigFile)) {
+                            echo "Cleansing file '${javaEnvironmentConfigFile}'"
+                            sh "rm -f ${javaEnvironmentConfigFile}"
+                        }
+                    }
+                }
+
+                echo "Stage result: clean: ${currentBuild.currentResult}"
+            }   
+        }
+    
+        stage('prepare deps') {
+            options {
+                timeout(time: 30, unit: 'MINUTES')  // Typically takes less than 2 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    sh "make prepare-deps CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                }
+
+                echo "Stage result: prepare deps: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage('build') {
+            options {
+                timeout(time: 6, unit: 'HOURS')     // Typically takes less than 2 hours
+            }
+            steps {
+                dir("$XADir") {
+                    script {
+                        if (isPR) {
+                            echo "PR build definition detected: building with 'make all'"
+                            sh "make all CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                        } else {
+                            echo "PR build definition *not* detected: building with 'make jenkins'"
+                            sh "make jenkins CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                        }
+                    }
+                }
+
+                echo "Stage result: build: ${currentBuild.currentResult}"
+            }
+        }
+        
+        stage('create vsix') {
+            options {
+                timeout(time: 30, unit: 'MINUTES')  // Typically takes less than 5 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    sh "make create-vsix CONFIGURATION=${env.BuildFlavor} V=1"
```

Comment:
This doesn't need `V=1`.

Reply:
Fixed with commit https://github.com/xamarin/xamarin-android/commit/0b5e0b3156555e768b98c3cc3e315eaf517e6d6c